editor/src/components/inspector/common/property-path-hooks.ts

```diff
@@ -376,31 +375,49 @@ function parseFinalValue<PropertiesToControl extends ParsedPropertiesKeys>(
 ): {
   finalValue: ParsedPropertiesValues
   isUnknown: boolean
+  usesComputedFallback: boolean
 } {
   const valueAsMaybe = eitherToMaybe(simpleValue)
   const rawValueAsMaybe = eitherToMaybe(rawValue)
 
-  function finalValueFromReal(): ParsedPropertiesValues {
-    if (realValue == null) {
-      const parsedComputedValue = parseAnyParseableValue(property, computedValue, null)
-      return defaultEither(emptyValues[property], parsedComputedValue)
-    } else {
-      const parsedRealValue = parseAnyParseableValue(property, realValue, null)
-      return defaultEither(emptyValues[property], parsedRealValue)
+  function finalValueFromReal(
+    isUnknown: boolean,
+  ): {
+    finalValue: ParsedPropertiesValues
+    isUnknown: boolean
+    usesComputedFallback: boolean
+  } {
+    const parsedRealValue = parseAnyParseableValue(property, realValue, null)
+    if (isRight(parsedRealValue)) {
+      return {
+        finalValue: parsedRealValue.value,
+        isUnknown: isUnknown,
+        usesComputedFallback: false,
+      }
     }
+    const parsedComputedValue = parseAnyParseableValue(property, computedValue, null)
```

Comment:
Can you turn these into `else if` statements please so it's immediately clear to see where the logic forks